controller/scheduler/mesos_marathon.py

```diff
@@ -0,0 +1,111 @@
+import re
+import time
+from django.conf import settings
+from marathon import MarathonClient
+from marathon.models import MarathonApp
+from .states import JobState
+from docker import Client
+from .fleet import FleetHTTPClient
+
+# turn down standard marathon logging
+
+MATCH = re.compile(
+    '(?P<app>[a-z0-9-]+)_?(?P<version>v[0-9]+)?\.?(?P<c_type>[a-z-_]+)?.(?P<c_num>[0-9]+)')
+RETRIES = 3
+
+
+class MarathonHTTPClient(object):
+
+    def __init__(self, target, auth, options, pkey):
+        self.target = settings.MARATHON_HOST
+        self.auth = auth
+        self.options = options
+        self.pkey = pkey
+        self.registry = settings.REGISTRY_HOST + ':' + settings.REGISTRY_PORT
+        self.client = MarathonClient('http://'+self.target+':8180')
+        self.fleet = FleetHTTPClient('/var/run/fleet.sock', auth, options, pkey)
+
+    # helpers
+    def _app_id(self, name):
+        return name.replace('_', '.')
+
+    # container api
+    def create(self, name, image, command='', **kwargs):
+        """Create a container"""
+        app_id = self._app_id(name)
+        l = locals().copy()
+        l.update(re.match(MATCH, name).groupdict())
+        image = self.registry + '/' + image
+        mems = kwargs.get('memory', {}).get(l['c_type'])
+        m = 0
+        if mems:
+            mems = mems.lower()
+            if mems[-2:-1].isalpha() and mems[-1].isalpha():
+                mems = mems[:-1]
+            m = int(mems[:-1])
+        c = 0.5
+        cpu = kwargs.get('cpu', {}).get(l['c_type'])
+        if cpu:
+            c = cpu
+        cpu = kwargs.get('cpu', {}).get(l['c_type'])
+        self.client.create_app(app_id,
+                               MarathonApp(cmd="docker run --name "+name+" -P "+image+" "+command,
+                                           mem=m, cpus=c))
```

Comment:
This is a bit hard to read, and generally speaking string formatting is preferable to concatenation in python to avoid object copies. I would do this:
```python
marathon_app = MarathonApp("docker run --name {name} -P {image} {command}".format(**locals), mem=m, cpus=c)
self.client.create_app(app_id, marathon_app)
```